homeassistant/components/matter/binary_sensor.py

```diff
@@ -34,60 +33,70 @@ async def async_setup_entry(
 class MatterBinarySensor(MatterEntity, BinarySensorEntity):
     """Representation of a Matter binary sensor."""
 
-    entity_description: MatterBinarySensorEntityDescription
-
-    @callback
-    def _update_from_device(self) -> None:
-        """Update from device."""
-        self._attr_is_on = self.get_matter_attribute_value(
-            # We always subscribe to a single value
-            self.entity_description.subscribe_attributes[0],
-        )
-
-
-class MatterOccupancySensor(MatterBinarySensor):
-    """Representation of a Matter occupancy sensor."""
-
-    _attr_device_class = BinarySensorDeviceClass.OCCUPANCY
-
     @callback
     def _update_from_device(self) -> None:
         """Update from device."""
-        value = self.get_matter_attribute_value(
-            # We always subscribe to a single value
-            self.entity_description.subscribe_attributes[0],
-        )
+        value: bool | uint | int | Nullable | None
+        value = self.get_matter_attribute_value(self._entity_info.primary_attribute)
+        if value in (None, NullValue):
+            value = None
+        elif value_convert := self._entity_info.measurement_to_ha:
+            value = value_convert(value)
+        self._attr_is_on = value
+
+
+# Discovery schema(s) to map Matter Attributes to HA entities
+DISCOVERY_SCHEMAS = [
+    # device specific: translate Hue motion to sensor to HA Motion sensor
+    # instead of generic occupancy sensor
+    MatterDiscoverySchema(
+        platform=Platform.BINARY_SENSOR,
+        entity_description=BinarySensorEntityDescription(
+            key="HueMotionSensor",
+            device_class=BinarySensorDeviceClass.MOTION,
+            name="Motion",
+        ),
+        entity_class=MatterBinarySensor,
+        required_attributes=(clusters.OccupancySensing.Attributes.Occupancy,),
+        vendor_id=(4107,),
+        product_name=("Hue motion sensor",),
+        measurement_to_ha=lambda x: (x & 1 == 1) if x is not None else None,
```

Comment:
How to convert to the entity state belongs in the entity description, not in the discovery schema.

Reply:
I was not sure if that is officially supporte din the entity description. If it is we'll have to adjust in a follow up PR

Reply:
If it's optional, just create a custom entity description class that inherits the platform entity description class and add the attribute. If it's required, create a mixin with just the required attribute and inherit from both the platform description and the mixin in custom entity description class.

Reply:
https://github.com/home-assistant/core/pull/90368